### PR TITLE
Feature/atom textarea

### PR DESCRIPTION
**Pull request by juanmaguitar**
`AtomTextarea` component according to the following specifications → https://paper.dropbox.com/doc/SUI-Text-area-L9OUNW4OHhcLmAEeFAbzR

### CODE REVIEW

**Comment by juanmaguitar** [comment]
Can be checked online at → https://sui-components-feature-atom-textarea.now.sh/workbench/atom/textarea/demo

File: components/atom/textarea/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  "description": "",
+  "main": "lib/index.js",
+  "scripts": {
+    "build": "npx rimraf ./lib && npx mkdirp ./lib && npm run build:js && npm run build:styles",
+    "build:js": "../../../node_modules/.bin/babel --presets sui ./src --out-dir ./lib",
+    "build:styles": "../../../node_modules/.bin/cpx './src/**/*.scss' ./lib"
+  },
+  "dependencies": {
+    "@s-ui/component-dependencies": "1",
+    "@s-ui/react-atom-help-text": "^1.1.0",
```
Comment (davidbarna) [inline components/atom/textarea/package.json]: use major versions

File: components/atom/textarea/src/index.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
+import React, {Fragment, Component} from 'react'
+import PropTypes from 'prop-types' // eslint-disable-line no-unused-vars
```
Comment (davidbarna) [inline components/atom/textarea/src/index.js]: what do you need `// eslint-disable-line no-unused-vars` for ?

File: components/atom/textarea/src/index.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+import cx from 'classnames'
+
+import AtomHelpText from '@s-ui/react-atom-help-text'
+import AtomValidationText, {
+  AtomValidationTextTypes
+} from '@s-ui/react-atom-validation-text'
+import AtomLabel, {AtomLabelTypes} from '@s-ui/react-atom-label'
+
+const BASE_CLASS = 'sui-AtomTextarea'
+const SIZES = {
```
Comment (davidbarna) [inline components/atom/textarea/src/index.js]: kind of weird to me to use "short" "long" when we're using "big" "small" in other components.
But I know it's the vocabulary written in the spec ^^

File: components/atom/textarea/src/index.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+  handleChange = e => {
+    const value = e.target.value
+    if (value.length > this.props.maxCharacters) return
+    this.setState({value})
+  }
+
+  get helpTextContent() {
+    const numCharacters = this.state.value ? this.state.value.length : 0
+    return `${numCharacters}/${this.props.maxCharacters} characters`
```
Comment (davidbarna) [inline components/atom/textarea/src/index.js]: `characters` should be translatable.

File: components/atom/textarea/src/index.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+AtomTextarea.displayName = 'AtomTextarea'
+
+AtomTextarea.propTypes = {
+  /** Text to be displayed as label of the textarea */
+  label: PropTypes.string.isRequired,
+
+  /** Size of button: 'short', 'long' */
+  size: PropTypes.oneOf(Object.values(SIZES)),
+
+  /** Sucess state of the component */
```
Comment (davidbarna) [inline components/atom/textarea/src/index.js]: typo success

File: components/atom/textarea/src/index.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+  /** Error state of the component */
+  error: PropTypes.bool,
+
+  /** Sucess message to display when sucess state  */
+  successText: PropTypes.string,
+
+  /** Error message to display when error state  */
+  errorText: PropTypes.string,
+
+  /** Maximum nomber of characters allowed  */
```
Comment (davidbarna) [inline components/atom/textarea/src/index.js]: typo number

File: components/atom/textarea/src/index.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  /** Size of button: 'short', 'long' */
+  size: PropTypes.oneOf(Object.values(SIZES)),
+
+  /** Sucess state of the component */
+  success: PropTypes.bool,
+
+  /** Error state of the component */
+  error: PropTypes.bool,
+
+  /** Sucess message to display when sucess state  */
```
Comment (davidbarna) [inline components/atom/textarea/src/index.js]: typo success

File: components/atom/textarea/src/index.scss [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+$fz-atom-textarea: $fz-base !default;
+$lh-atom-textarea: $lh-l !default;
+
+.sui-AtomTextarea {
+  font-size: $fz-atom-textarea;
+  padding-left: $p-l;
+  padding-top: $pm-atom-textarea;
+  line-height: $lh-atom-textarea;
+
+  &[disabled] {
```
Comment (davidbarna) [inline components/atom/textarea/src/index.scss:17]: Usually I use only class selectors for performance concern. But if you're sure it's not a problem here it's fine to me.

File: demo/atom/textarea/playground [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        padding: '10px'
+      }}
+    >
+      <AtomTextarea  label="Description" success successText="Perfect!">
+        Saul Bass on failure: Failure is built into creativity
+      </AtomTextarea>
+    </div>
+    <h2>Error</h2>
+    <div
+      style={{
```
Comment (davidbarna) [inline demo/atom/textarea/playground]: To make your demo more readable, you should store these common styles in a const

**Review by jcger — APPROVED** [review]

File: components/atom/textarea/src/index.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    const value = ev.target.value
+    const {onChange, maxCharacters} = this.props // eslint-disable-line react/prop-types
+    if (value.length > maxCharacters) return
+    this.setState({value}, () => onChange && onChange({value, ev}))
+  }
+
+  render() {
+    const {onChange, maxCharacters, size, ...props} = this.props
+    return (
+      <Fragment>
```
Comment (jcger) [inline components/atom/textarea/src/index.js]: you can remove the Fragment

File: components/atom/textarea/src/index.js [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+AtomTextarea.displayName = 'AtomTextarea'
+
+AtomTextarea.propTypes = {
+  /** Size of button: 'short', 'long' */
+  size: PropTypes.oneOf(Object.values(SIZES)),
+
+  /** Maximum number of characters allowed  */
+  maxCharacters: PropTypes.number.required
```
Comment (davidmartin84) [inline components/atom/textarea/src/index.js]: You don't need "required" in this prop because this props has defaultProps

**Review by davidmartin84 — APPROVED** [review]

**Pull request merged**